src/plugins/intel_cpu/src/nodes/unique.cpp

```diff
@@ -0,0 +1,399 @@
+    Node::createPrimitive();
+}
+
+void Unique::prepareParams() {
+    auto& dataMemPtr = getParentEdgeAt(IN_DATA)->getMemoryPtr();
+    if (!dataMemPtr || !dataMemPtr->isAllocated()) {
+        THROW_ERROR << " has not allocated input data memory.";
+    }
+    for (int i = 0; i < 4; i++) {
+        if (definedOutputs[i]) {
+            auto& dstMemPtr = getChildEdgeAt(i)->getMemoryPtr();
+            if (!dstMemPtr || !dstMemPtr->isAllocated()) {
+                THROW_ERROR << " has not allocated output memory at port " << i;
+            }
+        }
+    }
+    if (getSelectedPrimitiveDescriptor() == nullptr) {
+        THROW_ERROR << " has unidentified preferable primitive descriptor.";
+    }
+}
+
+void Unique::execute(dnnl::stream strm) {
+    if (flattened) {
+        size_t uniqueLen = 1lu;
+        if (dataPrecision == Precision::FP32) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::FP32>::value_type>();
+        } else if (dataPrecision == Precision::I32) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::I32>::value_type>();
+        } else if (dataPrecision == Precision::I8) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::I8>::value_type>();
+        } else if (dataPrecision == Precision::U8) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::U8>::value_type>();
+        }
+
+        const size_t srcLen = getParentEdgeAt(IN_DATA)->getMemoryPtr()->GetSize() / dataPrecision.size();
+        redefineOutputMemory({ {uniqueLen}, {uniqueLen}, {srcLen}, {uniqueLen}});
+    } else {
+        size_t uniqueLen = 1lu;
+        if (dataPrecision == Precision::FP32) {
+            uniqueLen = slicedTensorExec<PrecisionTrait<Precision::FP32>::value_type>();
+        } else if (dataPrecision == Precision::I32) {
+            uniqueLen = slicedTensorExec<PrecisionTrait<Precision::I32>::value_type>();
+        } else if (dataPrecision == Precision::I8) {
+            uniqueLen = slicedTensorExec<PrecisionTrait<Precision::I8>::value_type>();
+        } else if (dataPrecision == Precision::U8) {
+            uniqueLen = slicedTensorExec<PrecisionTrait<Precision::U8>::value_type>();
+        }
+
+        auto dstDataShape = getParentEdgeAt(IN_DATA)->getMemoryPtr()->getStaticDims();
+        const size_t srcLen = dstDataShape[axis];
+        dstDataShape[axis] = uniqueLen;
+        redefineOutputMemory({ dstDataShape, {uniqueLen}, {srcLen}, {uniqueLen}});
+    }
+}
+
+void Unique::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+template <typename T>
+size_t Unique::flattenTensorExec() {
+    const T* srcDataPtr = reinterpret_cast<const T*>(getParentEdgeAt(IN_DATA)->getMemoryPtr()->GetPtr());
+    const int64_t inputLen = getParentEdgeAt(IN_DATA)->getMemoryPtr()->GetSize() / sizeof(T);
+    T* uniqueData = reinterpret_cast<T*>(getChildEdgesAtPort(UNIQUE_DATA)[0]->getMemoryPtr()->GetPtr());
+    int *firstPtr, *inToOutPtr, *occurPtr;
```

Comment:
uninitialized variance is used, which result "warning as error" when building.

Reply:
Done